cmd/command/recovery.go

```diff
@@ -85,22 +89,46 @@ func recoverObjectAction(ctx *cli.Context) error {
 	}
 
 	if ctx.IsSet(objectListFlag.Name) {
-		if ctx.NArg() < 1 {
-			return fmt.Errorf("args number error")
-		}
-		argNum := ctx.NArg()
-		for i := 0; i < argNum-1; i++ {
-			recoverObject(bucketName, ctx.Args().Get(i), cfg, client)
+		objectNames := ctx.String(objectListFlag.Name)
+		objectNameList := strings.Split(objectNames, "//")
+		for _, objectName := range objectNameList {
+			recoverObject(bucketName, objectName, cfg, client)
 		}
 	} else {
 		objectName := ctx.String(objectFlag.Name)
 		recoverObject(bucketName, objectName, cfg, client)
 	}
+
+	taskCheckTicker := time.NewTicker(RecoveryJobSyncTime)
+
+	for range taskCheckTicker.C {
+		stats, _ := client.GetTasksStats(context.Background())
+		if stats.GetRecoveryProcessCount() > 0 {
+			log.Infof("Processing recovery tasks, update every 5 seconds. Waiting Tasks Number: %d. Recovered: %s. Failed: %s", stats.GetRecoveryProcessCount(), stats.GetRecoverySucceedList(), stats.GetRecoveryFailedList())
+		} else {
+			log.Infof("Finished process all recovery objects task on background. Recovered: %s. Failed: %s \n", stats.GetRecoverySucceedList(), stats.GetRecoveryFailedList())
+			break
+		}
+	}
+
 	return nil
 }
 
 func recoverObject(bucketName string, objectName string, cfg *gfspconfig.GfSpConfig, client *gfspclient.GfSpClient) error {
-	var replicateIdx int
+	var (
+		replicateIdx    int
+		processingCount uint32
+	)
+
+	stats, _ := client.GetTasksStats(context.Background())
+	processingCount = stats.GetRecoveryProcessCount()
+	for processingCount >= MaxRecoveryJob {
```

Comment:
There may be multiple users and multiple threads calling the cmd recover command, and limiting the number of jobs called at a time may not have much effect.

Reply:
The stats.GetRecoveryProcessCount() get the shared recoveryTaskMap from manager module,  and recoveryTaskMap will be shared across threads, so it shall give the real time jobs waiting for process.